pkg/trace/agent/agent.go

```diff
@@ -296,11 +296,18 @@ func (a *Agent) ProcessStats(in pb.ClientStatsPayload, lang, tracerVersion strin
 	in.TracerVersion = tracerVersion
 	in.Lang = lang
 	for _, group := range in.Stats {
+		n := 0
 		for _, b := range group.Stats {
```

Comment:
Can we please test this thoroughly in unit tests? We are rewriting the same slice that we are looping over. I'm not aware of the intricacies of this and whether it could be a problem or not?